### ISSUE TITLE
"Double translation" of names

### ISSUE THREAD

**Issue by Maturion** [mapsme/omim#649]
In OSM, some countries that have an official language with a non-Latin alphabet are mapped in that language only, while in other countries a Latin transcription is added to all default name tags. The most promiment examples are South Korea and Georgia: 

Places are tagged liked this: 
name ="თბილისი (Tbilisi)"
name:ka="თბილისი" 
name:en="Tbilisi"

Maps.me currently doesn't detect that and is therefore showing double translations of names.
My phone language is set to German, and Tbilisi is shown to me as "თბილისი (Tbilisi) Tiflis", while a Russian user would get "თბილისი (Tbilisi) Тбилиси".

That translation in brackets should be cut off by Maps.me in order to avoid these redundant translations.

**Comment by deathbaba** [comment]
If you are speaking about (Tbilisi) part in a name tag, then it should be fixed in OSM name tag itself, not in MAPS.ME.

**Comment by Maturion** [comment]
The Georgian mapping community has explicitly decided to use this naming pattern for every object in the whole country (Georgian name + English name in brackets). Same is true for South Korea. Therefore this cannot be fixed in OSM, it's a conscious decision by the local mapping community.

MapBox is also cutting off the brackets in those countries.

**Comment by twzkxkan** [comment]
I agree with @deathbaba, this is an OSM data issue.
>The Georgian mapping community has explicitly decided 

I think you'll find that OpenStreetMap works best when participants agree to a consensus. I'm sure there is a real need to display both languages, and that can be done by choosing which tags to render, not by adding incorrect data.

**Comment by Maturion** [comment]
Yes, I fully agree to that. It's all up to local users. However, this leads to strange rendering of names in Maps.me. Even a user that has set his language to English will see the English name twice then (თბილისი (Tbilisi) Tbilisi), making Maps.me's localisation pointless in that country. It's only a couple of lines of code to fix this.

At MapBox they are aware of this situation and they're handling it by cutting the name in brackets.

**Comment by twzkxkan** [comment]
I am considering unnecessary to add special code useful only to improve display of invalid data.
Besides, it only encourages bad tagging.

**Comment by Maturion** [comment]
I wouldn't consider that "invalid data". Mappers chose to deliberately add these brackets and there's a system behind that. In South Korea, the naming pattern (Korean name + English name in brackets) is the same. This is not going to change anytime soon.

Also, there are many ways to prevent cutting of  normal use of parentheses. You could simply check if the name inside brackets and name:en are identical/close to identical.

 I consider this to be a valid issue.

**Comment by twzkxkan** [comment]
>Mappers chose to deliberately add these brackets

See [tagging for the renderer](https://wiki.openstreetmap.org/wiki/Tagging_for_the_renderer).

>You could simply check if the name inside brackets and name:en are identical/close to identical.

It's not simple nor practical. IMHO it's unreasonable to change every OSM software to accomodate conflicting tag use by a small fraction of the world. Especially when the same result can be achieved with valid data.

A good place to continue this discussion is on the [tagging mailing list](https://lists.openstreetmap.org/listinfo/tagging) and those local mapping communities.

**Comment by Maturion** [comment]
Well, as far as I know, every local mapping decides on its own how to tag things in their respective area.  Therefore I don't see the situation in Georgia and/or South Korea as an example of invalid data. It's [the way they chose](https://wiki.openstreetmap.org/wiki/WikiProject_Georgia_Abkhazia_South-Ossetia) to map their country. Asking them to change this would be "tagging for the renderer" IMHO. 

The relevant code to fix this issue is in omim/indexer/feature.php, inside `struct BestMatchedLangNames`. This struct already contains code to fix specific tagging issues in some Arab countries, so I personally don't see why there shouldn't be code to fix those issues as well. 

But still, that's just my opinion.

**Comment by deathbaba** [comment]
Two points to keep in mind about any code change related to this issue:

1. BestMatchedLangNames is a runtime check which slows down ALL processed strings for ANY feature on a map, so it is a bad place for this fix. Probably check at data generation stage should be much better, as it is possible to find a matching substring and cut name tag with a server power.
2. We’re planning to launch POI editor inside MAPS.ME. After this “fix”, what should user see on a Edit screen? Original name tag with (English) name? Or cut-off string? But If user (who do not necessary knows about parentheses) edit this cut-off name, it will overwrite an old one with parentheses.

P.S. Seriously, it would be much easier for everyone, if data will follow common standard, and OSM users propose fixes to renderers about displaying name in their local language + international one.

**Comment by mxa** [comment]
@Maturion for South Korea I can say that this problem was acknowledged and currently the tags are in the process of being changed.  It's a very long process. See [OSM Wiki](https://wiki.openstreetmap.org/wiki/Ko:Map_Features#.ED.91.9C.EA.B8.B0.EB.B2.95)
